packages/client-react-streaming/src/registerApolloClient.tsx

```diff
@@ -29,34 +37,69 @@ const seenClients = WeakSet
  *
  * @public
  */
-export function registerApolloClient(
-  makeClient: () => Promise<ApolloClient<any>>
-): { getClient: () => Promise<ApolloClient<any>> };
-/**
- * Ensures that you can always access the same instance of ApolloClient
- * during RSC for an ongoing request, while always returning
- * a new instance for different requests.
- *
- * @example
- * ```ts
- * export const { getClient } = registerApolloClient(() => {
- *   return new ApolloClient({
- *     cache: new InMemoryCache(),
- *     link: new HttpLink({
- *       uri: "http://example.com/api/graphql",
- *     }),
- *   });
- * });
- * ```
- *
- * @public
- */
-export function registerApolloClient(makeClient: () => ApolloClient<any>): {
-  getClient: () => ApolloClient<any>;
-};
-export function registerApolloClient(
-  makeClient: (() => Promise<ApolloClient<any>>) | (() => ApolloClient<any>)
-) {
+export function registerApolloClient<
+  AC extends Promise<ApolloClient<any>> | ApolloClient<any>,
```

Comment:
[nit] As a small thing, could we be a bit more verbose with the TS name here so its easily recognizable what `AC` is?

How about
```suggestion
  TApolloClient extends Promise<ApolloClient<any>> | ApolloClient<any>,
```
or
```suggestion
  ApolloClientInstance extends Promise<ApolloClient<any>> | ApolloClient<any>,
```

(I personally like the `TApolloClient`, but I'll leave it up to you.

Reply:
what about `ApolloClientOrPromise`?

Reply:
[b162c52](https://github.com/apollographql/apollo-client-nextjs/pull/258/commits/b162c52b4eafcede69e7b8388e03c571b1ce8293)